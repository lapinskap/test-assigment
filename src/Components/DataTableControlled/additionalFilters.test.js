import React from 'react';
import renderer from 'react-test-renderer';
import {
  render, fireEvent,
} from '@testing-library/react';
import AdditionalFilters from './additionalFilters';
import '../../tests/setupTests';

describe('<AdditionalFilters />', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly with props', () => {
    const tree = renderer
      .create((
        <AdditionalFilters
          onFilteredChange={() => {
          }}
          currentFilters={[{ id: 'test_select', value: '2' }]}
          filtersConfig={[
            {
              type: 'text',
              label: 'Test Text Filter',
              id: 'test_text',
            },
            {
              type: 'select',
              label: 'Test Select Filter',
              id: 'test_select',
              options: [{
                value: '1',
              }, {
                label: 'Option 2',
                value: '2',
              }],
            },
          ]}
        />
      ))
      .toJSON();

    expect(tree)
      .toMatchSnapshot();
  });

  it('correctly changed filters', async () => {
    const onFilteredChangeSpy = jest.fn();
    const { container } = render((
      <AdditionalFilters
        onFilteredChange={onFilteredChangeSpy}
        currentFilters={[{ id: 'test_select', value: '2' }]}
        filtersConfig={[
          {
            type: 'text',
            label: 'Test Text Filter',
            id: 'test_text',
          },
          {
            type: 'select',
            label: 'Test Select Filter',
            id: 'test_select',
            options: [{
              value: '1',
            }, {
              label: 'Option 2',
              value: '2',
            }],
          },
        ]}
      />
    ));

    const input = container.querySelector('input');
    fireEvent.change(input, { target: { value: '1' } });
    expect(onFilteredChangeSpy).toHaveBeenCalledTimes(1);

    const select = container.querySelector('select');
    fireEvent.change(select, { target: { value: '1' } });
    expect(onFilteredChangeSpy).toHaveBeenCalledTimes(2);
  });
});
