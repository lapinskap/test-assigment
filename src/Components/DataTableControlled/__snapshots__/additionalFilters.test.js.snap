// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdditionalFilters /> renders correctly with props 1`] = `
<div
  className="row"
>
  <div
    className="mb-2 col-sm-2"
  >
    <label
      className=""
    >
      Test Text Filter
      :
    </label>
     
    <input
      className="form-control"
      data-t1="gridFilter"
      data-t2="test_text"
      id="test_text"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="mb-2 col-sm-2"
  >
    <label
      className=""
    >
      Test Select Filter
      :
    </label>
     
    <select
      className="form-control"
      data-t1="gridFilter"
      data-t2="test_select"
      id="test_select"
      onChange={[Function]}
      value="2"
    >
      <option
        value="1"
      >
        1
      </option>
      <option
        value="2"
      >
        Option 2
      </option>
    </select>
  </div>
</div>
`;
